partner_tier_validation/readme/DESCRIPTION.rst

```diff
@@ -0,0 +1,12 @@
+Adds an approval workflow to Partners.
+The default rule requires new company Contacts to be approved
+before they can be used.
+
+The rule can be extended to new non-company contact,
+but beware that may cause issues with automatically created new contacts,
+such as the ones generated when processing incoming emails.
+
+If the 'Is Company' or 'Parent' field changes then the contact is Request
+for approval.
+
+For this, the new Contact record is kept as "Archived" until it is approved.
```

Comment:
I believe so. It sounds like that, as is, this module might not make much sense?
Maybe it could enforce that only approved records are approved, encapsulating that decision in a computed field.
The Partner Stage related module could then extend that for richer state-based decisions.